maven-core/src/main/java/org/apache/maven/graph/DefaultProjectDependencyGraph.java

```diff
@@ -61,39 +61,25 @@
      * @throws DuplicateProjectException
      * @throws CycleDetectedException
      */
-    public DefaultProjectDependencyGraph( Collection<MavenProject> projects )
-        throws CycleDetectedException, DuplicateProjectException
+    public DefaultProjectDependencyGraph( List<MavenProject> projects )
```

Comment:
Attention: You are changing a public contract here. Is this intended?

Reply:
IMO Maven uses List too often where Collection should have been used. I'm pretty sure this needs to be reverted.

Reply:
If I don't do that I have to cast explicit...
```java
    public DefaultProjectDependencyGraph( Collection<MavenProject> projects )
            throws CycleDetectedException, DuplicateProjectException
    {
        this( (List<MavenProject>) projects, projects );
    }
```
Furthermore the current calls to ctor are only using List...

Reply:
The cast is dangerous because you don't know that this is a list actually. Why not wrap it in a list, idiotproof?!

Reply:
`this(new ArrayList(projects), projects);` ?

Reply:
Ok so let us go that direction:
```java
    public DefaultProjectDependencyGraph( Collection<MavenProject> allProjects,
                                          Collection<MavenProject> projects )
                                          ...
  ...}
                                          
    public DefaultProjectDependencyGraph( Collection<MavenProject> projects )
            throws CycleDetectedException, DuplicateProjectException
    {
        this( projects, projects );
    }
```

Reply:
> `this(new ArrayList(projects), projects);` ?

This would result in duplication of `new ArrayList<>(..)` because the called ctor starts with:
```java
    public DefaultProjectDependencyGraph( Collection<MavenProject> allProjects,
                                          Collection<MavenProject> projects )
            throws CycleDetectedException, DuplicateProjectException
    {
        this.allProjects = Collections.unmodifiableList( new ArrayList<>( allProjects ) );
        this.sorter = new ProjectSorter( projects );
        this.order = new HashMap<>();
...
I think the easiest way is to change both constructors to use Collection which will solve that...